src/app/services/tab.service.ts

```diff
@@ -58,7 +59,20 @@ export class TabService implements OnDestroy {
   }
 
   setActiveTab(tag: string|undefined): void {
-    this.activeTab.next(tag);
+    const destroyed$ = new Subject<void>();
+    this.router.events.pipe(
+      tap(event => {
+        if (event instanceof NavigationCancel) {
+          destroyed$.next();
+          destroyed$.complete();
+        }
+      }),
+      filter(event => event instanceof NavigationEnd),
+      take(1),
+      takeUntil(destroyed$),
+    ).subscribe(() =>
+      this.activeTab.next(tag)
+    );
```

Comment:
Will `destroyed$` be fully released if there is no navigation cancel event?  I am not sure.

Reply:
Fixed